File: api/apps/finance/internal/app/stripe.go
package app

import (
	"kloudlite.io/apps/finance/internal/domain"
	"kloudlite.io/pkg/stripe"
)

type pStripe struct {
	stripeCli *stripe.Client
}

func (p pStripe) MakePayment(customerId stripe.CustomerId, paymentMethodId string, amount float64) (*stripe.Payment,
	error) {
	return p.stripeCli.NewPayment(customerId, paymentMethodId, amount)
}

func (p pStripe) CreateCustomer(accountId string, paymentMethodId string) (*stripe.CustomerId, error) {
	return p.stripeCli.NewCustomer(accountId, paymentMethodId)
}

func (p pStripe) GetSetupIntent() (string, error) {
	return p.stripeCli.NewSetupIntent()
}

func NewStripeClient(env *Env) domain.Stripe {
	return &pStripe{stripeCli: stripe.NewClient(env.StripeSecretKey)}
}

File: web/apps/console/src/lib/stripe.ts
import Stripe from 'stripe'

let instance: Stripe | null = null

export function getStripe(): Stripe {
  if (!instance) {
    const secretKey = process.env.STRIPE_SECRET_KEY
    if (!secretKey) {
      throw new Error('STRIPE_SECRET_KEY must be set')
    }
    instance = new Stripe(secretKey)
  }
  return instance
}
